### ISSUE TITLE
Bug: no-unused-vars incorrectly reports warnings for nested variables

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#296]
The following example will report a warning for variable `test`:

```
function a() {
  var test = 0;
  (function() {
    test = test + 1;
  })();
  return test;
};
a();
```

This is due to the bug in escope where it incorrectly moves `test` from anonymous function into global context. I filed a bug for escope for this issue here: https://github.com/Constellation/escope/issues/23

This ticket is mostly for tracking purposes.
